src/app/Fake.Core.Target/Target.fs

```diff
@@ -644,40 +691,51 @@ module Target =
         if singleTarget
         then Trace.traceImportant "Single target mode ==> Skipping dependencies."
         let allTargets = List.collect Seq.toList order
-        let context = TargetContext.Create targetName allTargets args
-        
+        use cts = new CancellationTokenSource()    
+        let context = TargetContext.Create targetName allTargets args false cts.Token
+          
         let context =
-            // Figure out the order in in which targets can be run, and which can be run in parallel.
-            if parallelJobs > 1 && not singleTarget then
-                Trace.tracefn "Running parallel build with %d workers" parallelJobs
-
-                // always try to keep "parallelJobs" runners busy
-                ParallelRunner.runOptimal parallelJobs order context
-                //order
-                //    |> Seq.fold (fun context par -> runTargetsParallel parallelJobs par context) context
-            else
-                let targets = order |> Seq.collect id |> Seq.toArray
-                let lastTarget = targets |> Array.last
-                if singleTarget then
-                    runSingleTarget lastTarget context
+            let captureContext (f:'a->unit) = 
+                let ctx = Context.getExecutionContext()
+                (fun a -> 
+                    let nctx = Context.getExecutionContext()
+                    if ctx <> nctx then Context.setExecutionContext ctx
+                    f a)
+                      
+            let cancelHandler  = captureContext (handleUserCancelEvent cts)            
+            use __ = Console.CancelKeyPress |> Observable.subscribeOnce cancelHandler
+            
+            let context =
+                // Figure out the order in in which targets can be run, and which can be run in parallel.
+                if parallelJobs > 1 && not singleTarget then
+                    Trace.tracefn "Running parallel build with %d workers" parallelJobs
+
+                    // always try to keep "parallelJobs" runners busy
+                    ParallelRunner.runOptimal parallelJobs order context
+                    //order
+                    //    |> Seq.fold (fun context par -> runTargetsParallel parallelJobs par context) context
```

Comment:
I think we can remove this now as well as the implementation if it still exists (though I can see if you don't want to do it as part of this PR - I wouldn't mind if we already do changes here)

Reply:
(Take this with caution:) Maybe we even can unify this now with the other branch (though I'm not sure if that actually looks better)

Reply:
You mean remove the ParallelRunner.runOptimal method and the whole `if parallelJobs > 1 && not singleTarget` ?

Reply:
That would then remove the whole ParallelRunner as far as i can see.

Reply:
No, I meant the comments and `runTargetsParallel`

Reply:
OK, i can remove those. That confused me ;)